infra/feast-operator/internal/controller/services/util.go

```diff
@@ -145,9 +168,36 @@ func defaultOnlineStorePath(persistence *feastdevv1alpha1.OnlineStoreFilePersist
 	}
 	return DefaultOnlineStorePvcPath
 }
+
 func defaultRegistryPath(persistence *feastdevv1alpha1.RegistryFilePersistence) string {
 	if persistence.PvcConfig == nil {
 		return DefaultRegistryEphemeralPath
 	}
 	return DefaultRegistryPvcPath
 }
+
+func checkOfflineStoreDBStorePersistenceType(value string) error {
+	if slices.Contains(feastdevv1alpha1.ValidOfflineStoreDBStorePersistenceTypes, value) {
+		return nil
+	}
+	return fmt.Errorf("invalid DB store type %s for offline store", value)
+}
+
+func checkOnlineStoreDBStorePersistenceType(value string) error {
+	if slices.Contains(feastdevv1alpha1.ValidOnlineStoreDBStorePersistenceTypes, value) {
+		return nil
+	}
+	return fmt.Errorf("invalid DB store type %s for online store", value)
+}
+
+func (feast *FeastServices) getSecret(secretRef string) (*corev1.Secret, error) {
+	secret := &corev1.Secret{ObjectMeta: metav1.ObjectMeta{Name: secretRef, Namespace: feast.FeatureStore.Namespace}}
+	objectKey := client.ObjectKeyFromObject(secret)
+	if err := feast.Client.Get(feast.Context, objectKey, secret); err != nil {
+		if apierrors.IsNotFound(err) || err != nil {
+			return nil, fmt.Errorf("invalid secret %s for offline store", secretRef)
```

Comment:
maybe just return the error here but log an Error prior to the return
```suggestion
			logger := log.FromContext(feast.Context)
			logger.Error("invalid secret %s for offline store", secretRef)
			return nil, err
```

Reply:
Done